### PR TITLE
Refactor modulefiles

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [ ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

All currently supported module files have been updated to load file which specifies all common libraries.
I'm not sure what to do about `odin` and `s4.intel`.

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [ ] hera.intel
- [ ] hera.gnu
- [ ] orion.intel
- [ ] cheyenne.intel 
- [ ] cheyenne.gnu
- [ ] gaea.intel 
- [ ] jet.intel
- [ ] wcoss_cray
- [ ] wcoss_dell_p3

## Dependencies

No.

### CODE REVIEW

File: modulefiles/ufs_hera.gnu [hunk 1]
```diff
@@ -18,25 +18,7 @@ setenv CMAKE_Platform hera.gnu
 module use /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack
 module load hpc/1.1.0
 module load hpc-gnu/9.2.0
-module load hpc-mpich/3.3.2  
+module load hpc-mpich/3.3.2
```
Comment (aerorahul) [inline modulefiles/ufs_hera.gnu:21]: is `hpc-mpich/3.3.2` meta-module not loading `mpich/3.3.2`? If so, the next line should be removed.

File: tests/compile.sh [hunk 2]
```diff
@@ -54,7 +54,7 @@ hostname
 
 set +x
 if [[ $MACHINE_ID == macosx.* ]] || [[ $MACHINE_ID == linux.* ]]; then
-  source $PATHTR/modulefiles/${MACHINE_ID}/fv3
+  source $PATHTR/modulefiles/ufs_${MACHINE_ID}
```
Comment (aerorahul) [inline tests/compile.sh:59]: `MACHINE_ID` also contains compiler, which is a little misleading.
e.g. `MACHINE_ID` = `hera.gnu` or `hera.intel` or `orion.intel` or `wcoss_dell_p3`.
There is a lack of consistency on what `MACHINE_ID` is supposed to mean.
Sometimes it is `machine`.`compiler` and sometimes it is just `machine`.

**Review by aerorahul — COMMENTED** [review]
Most changes look good to me.
I have 2 comments:
1. see comment under `ufs_hera.gnu`
2. see comment regarding the use of `MACHINE_ID`.

**Comment by MinsukJi-NOAA** [comment]
I will add corresponding changes to the utest script.

File: tests/compile.sh (hunk 2 above)
Comment (climbfuji) [reply tests/compile.sh:59]: Oh yes, that is "historical", we have moved everything except the wcoss platforms to contain compiler information. I don't know why this didn't happen for wcoss. Agreed that machine_id is not the best name for it, but as long as we are consistent ... we could go through all scripts (and regression test files) and rename it to "platform" maybe? In a separate PR ...

**Review by climbfuji — APPROVED** [review]
I would just rename the odin s4 modulefiles as you did for the others. It's the same as for stampede2, it will work for a while when someone updates it, and then age off.

**Review by MinsukJi-NOAA — APPROVED** [review]

**Review by aerorahul — APPROVED** [review]
Looks good

File: tests/compile.sh (hunk 2 above)
Comment (junwang-noaa) [reply tests/compile.sh:59]: I think the changes of making consistent MACHINE_ID might not be trivial, it's better to do it in separate PR.

File: tests/utest [hunk 3]
```diff
@@ -428,16 +428,13 @@ fi
 # utility functions in rt_utils need to be able to see variables in utest
 source rt_utils.sh
 
+source $PATHTR/NEMS/src/conf/module-setup.sh.inc
 # Machine-dependent libraries, modules, variables, etc.
 if [[ $MACHINE_ID = hera.* ]]; then
 
-  source $PATHTR/NEMS/src/conf/module-setup.sh.inc
-  module use $PATHTR/modulefiles/${MACHINE_ID}
-  module load fv3
```
Comment (junwang-noaa) [inline tests/utest:436]: I am curious, do we not need to load the module file when running utest on hera?

Comment (MinsukJi-NOAA) [reply tests/utest:436]: Similar to rt.sh, modulefiles are loaded by compile.sh and run_test.sh.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request closed without merging**